vnext/ReactUWP/Views/ReactControl.cpp

```diff
@@ -339,6 +340,10 @@ void ReactControl::PrepareXamlRootView(XamlView const &rootView) {
     children.Clear();
 
     auto newRootView = winrt::Grid();
+    auto perspectiveTransform3D = winrt::Windows::UI::Xaml::Media::Media3D::PerspectiveTransform3D();
+    perspectiveTransform3D.Depth(850);
+    winrt::Windows::UI::Xaml::Media::Media3D::Transform3D t3d(perspectiveTransform3D);
+    newRootView.Transform3D(t3d);
```

Comment:
I think ideally this would be pay to play.  Here you'll have set perspective transforms on every element in the tree, which I think is expensive (I'm making an assumption here).

Reply:
@StephenLPeters  This is only in the root view object though, so it's one object only

Reply:
There is only one ReactControl per app instance?

---
In reply to: [340372307](https://github.com/microsoft/react-native-windows/pull/3551#discussion_r340372307) [](ancestors = 340372307)

Reply:
There can be multiple ReactControls / react root views per instance.  -- This is something that the current high level RNW Instance interface doesn't provide good APIs for, but its certainly possible, and something that's going to be better exposed in the future.

Reply:
When we expose multiple reactcontrols for an app as Andrew mentioned, we can think about how this can work :)

---
In reply to: [340373548](https://github.com/microsoft/react-native-windows/pull/3551#discussion_r340373548) [](ancestors = 340373548,340372307)

Reply:
To be clear, you can already have multiple reactcontrols and I believe we already have cases where we do that.

Having said that I wouldn't expect to have huge numbers of them.

Reply:
Thanks for the info :) Sounds like it's ok then to have this property set on each reactcontrol instance.

---
In reply to: [340425050](https://github.com/microsoft/react-native-windows/pull/3551#discussion_r340425050) [](ancestors = 340425050)

Reply:
Maybe ask george gao and keith melmon what the performance impact of a perspective on the root is? I'm worried about the cost of the perspective being applied to the whole tree

---
In reply to: [340373548](https://github.com/microsoft/react-native-windows/pull/3551#discussion_r340373548) [](ancestors = 340373548,340372307)